### PR TITLE
v2.1.0 RC

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #128, #140, #147, #150, #151, #153, #154, #156, #159, #173

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Create new slot that allows users to put content into each day cell in the calendar
* Style the content within the day slot
* Create examples to demonstrate all of our use cases for this slot (text and `auro-icon`)
* Update repo to support node 20
* Update design token prefixes from `--auro` to `--ds`
* Add icon bundled script to demo page in order for icons to render in the datepicker examples
* Update mobile gradient color to be able to see later calendars easily
* Refactor mobile design to implement 320px breakpoint, which in turn:
    - Removes the back button from mobile header
    - Makes calendar size responsive
    - Updates spacing between calendars and it's items
    - Updates size and text of mobile calendar button
* Add `auro-popover` to `auro-calendar-cell` with the ability to pass popover content from the `auro-datepicker` level
* Fix `centralDate` attribute functionality
* Refactor styles according to datepicker blueprint
* Update to use latest dependencies
* Add cssparts to datepicker components

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/auro-calendar.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import { html } from 'lit';
 import styleCss from "./style-auro-calendar-css";
 
+// lodash is being used to group the slot content by month
+import 'lodash';
+
```
Comment (jason-capsule42) [inline src/auro-calendar.js:6]: We aren't using this any more with the new method for moving slot content.

File: src/auro-datepicker.js [hunk 2]
```diff
@@ -649,16 +694,17 @@ export class AuroDatePicker extends LitElement {
   // function that renders the HTML and CSS into  the scope of the component
   render() {
     return html`
-      <div>
+      <div part="datepickerContainer">
```
Comment (jason-capsule42) [inline src/auro-datepicker.js:697]: What does this allow a user to style that they can't by using auro-dropdown part?

File: src/style-auro-calendar-cell.scss [hunk 3]
```diff
@@ -1,73 +1,175 @@
...
+
+    font-size: var(--ds-text-body-size-lg, $ds-text-body-size-lg);
+  }
+
+  .dateSlot {
+    font-size: var(--ds-text-body-size-xs, $ds-text-body-size-xs);
   }
 }
 
 @media (hover: hover) {
```
Comment (jason-capsule42) [inline src/style-auro-calendar-cell.scss:161]: Something is going on with the hover affect with range support - demoed live to @jordanjones243 earlier.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: src/auro-datepicker.js (hunk 2 above)
Comment (jordanjones243) [reply src/auro-datepicker.js:697]: Fair point, this is removed.

File: src/style-auro-calendar-cell.scss (hunk 3 above)
Comment (jordanjones243) [reply src/style-auro-calendar-cell.scss:161]: Fixed by adding a function in the `classMap` of `auro-calendar-cell` called `isInRange`.

**Review by leeejune — DISMISSED** [review]

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**